runtime/sema/crypto_algorithm_types.go

```diff
@@ -0,0 +1,190 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sema
+
+import (
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/errors"
+)
+
+//go:generate go run golang.org/x/tools/cmd/stringer -type=SignatureAlgorithm
+//go:generate go run golang.org/x/tools/cmd/stringer -type=HashAlgorithm
+
+var SignatureAlgorithms = []NativeEnumCase{
+	SignatureAlgorithmECDSA_P256,
+	SignatureAlgorithmECDSA_Secp256k1,
+	SignatureAlgorithmBLSBLS12381,
+}
+
+var HashAlgorithms = []NativeEnumCase{
+	HashAlgorithmSHA2_256,
+	HashAlgorithmSHA2_384,
+	HashAlgorithmSHA3_256,
+	HashAlgorithmSHA3_384,
+	HashAlgorithmKMAC128,
+}
+
+var SignatureAlgorithmType = newNativeEnumType(SignatureAlgorithmTypeName, &UInt8Type{})
+
+type SignatureAlgorithm int
+
+const (
+	// Supported signing algorithms
+	SignatureAlgorithmECDSA_P256 SignatureAlgorithm = iota
+	SignatureAlgorithmECDSA_Secp256k1
+	SignatureAlgorithmBLSBLS12381
+)
+
+// Name returns the string representation of this signing algorithm.
+func (algo SignatureAlgorithm) Name() string {
+	switch algo {
+	case SignatureAlgorithmECDSA_P256:
+		return "ECDSA_P256"
+	case SignatureAlgorithmECDSA_Secp256k1:
+		return "ECDSA_Secp256k1"
+	case SignatureAlgorithmBLSBLS12381:
+		return "BLSBLS12381"
+	}
+
+	panic(errors.NewUnreachableError())
+}
+
+func (algo SignatureAlgorithm) RawValue() int {
+	return int(algo)
+}
+
+func (algo SignatureAlgorithm) DocString() string {
+	switch algo {
+	case SignatureAlgorithmECDSA_P256:
+		return SignatureAlgorithmDocStringECDSA_P256
+	case SignatureAlgorithmECDSA_Secp256k1:
+		return SignatureAlgorithmDocStringECDSA_Secp256k1
+	case SignatureAlgorithmBLSBLS12381:
+		return SignatureAlgorithmDocStringBLSBLS12381
+	}
+
+	panic(errors.NewUnreachableError())
+}
+
+var HashAlgorithmType = newNativeEnumType(HashAlgorithmTypeName, &UInt8Type{})
+
+type HashAlgorithm int
+
+const (
+	// Supported hashing algorithms
+	HashAlgorithmSHA2_256 HashAlgorithm = iota
```

Comment:
I don't think there is a need to support the `384` versions in Cadence, unless there is another reason I'm not aware of. 
Should we remove them? cc @turbolent @psiemens

Reply:
I only add them because the FVM would probably support those algorithms (https://github.com/onflow/flow-go/blob/master/crypto/hash/types.go#L8:L16).

If are not planning to support them, I am all-in to remove them.